navapbc/template-infra
# Update environment backend configuration templatees

## Ticket

n/a

## Changes

* Move environment modules to app/envs
  This provides more flexibility to projects who have multiple apps per rep
  with no real disadvantage
* Update environment module backend configurations to have easier template properties
* Add setup-app-backends script for testing out backend configuration setup

## Context for reviewers

I wanted to update the environment module backend configurations to have template parameters that are easier to replace programmatically, so I surrounded template variables `<>` brackets. This will hopefully make it easier to do e2e testing for the template.

I haven't figured out how to automate the setup for multi-account setups, but I think that can be out of scope for this PR.

## Testing

* Run `make -f template-only.mak bootstrap-account`
* Run `make -f template-only.mak setup-app-backends`

Check the environment main.tf file backend configurations.
<img width="640" alt="image" src="https://user-images.githubusercontent.com/447859/195467960-f6e9c3d9-636f-4a6d-a313-a0d91842f642.png">

Here's an example for prod
<img width="1460" alt="image" src="https://user-images.githubusercontent.com/447859/195467933-46ec04e0-3838-4775-bd48-d8b28e676d6b.png">
